src/features/instrument/Instrument.tsx

```diff
@@ -0,0 +1,155 @@
+import { Button } from '@mui/material';
+import './Instrument.css';
+import { Synth, Volume, Sampler, Destination, PolySynth, Transport, Part, Noise, AutoFilter, Player } from 'tone'
+import { BASS_CHORDS, GAMMA_NOTES, MARIO_SAMPLES, MELODY_CHORDS } from '../../constants/notes'
+import { useRef } from 'react';
+const DogSample = require("../../assets/dog-sample.mp3");
+
+const keyStyles = {
+  paddingTop: '4em',
+  paddingBottom: '4em',
+  width: '5em'
+};
+const mediumVolume = new Volume(-15);
+const sampler = new Sampler({
+  E4: DogSample,
+}).chain(mediumVolume, Destination);
+const polySynthSquare = new PolySynth(Synth, {
+  oscillator: {
+    type: "square",
+  },
+}).chain(new Volume(-25), Destination);
```

Comment:
initialization of constants
Brace yourself for a long comment :D  I put much time on this and need your help to clarify it.
There are values `mediumVolume`, `sampler`, `polySynthSquare` which I wanted to initialize for my component. Those are instruments I am using in other places to play something. There are several ways I made that work, my thinking flow was like that:

1st way - add simple as const variables inside Instrument
`const mediumVolume = new Volume(-15);
  const sampler = new Sampler({
    E4: DogSample,
  }).chain(mediumVolume, Destination);

  const polySynthSquare = new PolySynth(Synth, {
    oscillator: {
      type: "square",
    },
  }).chain(new Volume(-25), Destination);`
* I know such declaration of variables doesn't preserve same value between rerenderings, but in my case that was not an issue: 
  1) same value is always reset (should I in general worry about the reassignment of constant (without any computation))
  2) no re-renderings happen in my app
*doesn't cause rerendering
  But, I was not sure, since here I am initializing not a primitive constant, but object, so was thinking maybe I need to use smth else as hooks, since that's what most commonly I saw in other code examples

2nd way - use useState inside Instrument
`const [mediumVolume] = useState(new Volume(-15));
  const [sampler] = useState(new Sampler({
    E4: DogSample,
  }).chain(mediumVolume, Destination));
  const [polySynthSquare] = useState(new PolySynth(Synth, {
    oscillator: {
      type: "square",
    },
  }).chain(new Volume(-25), Destination));`
  
  3rd way - use useRef
  `const mediumVolume = useRef(new Volume(-15)).current;
  const sampler = useRef(new Sampler({
    E4: DogSample,
  }).chain(mediumVolume, Destination)).current;
  const polySynthSquare = useRef(new PolySynth(Synth, {
    oscillator: {
      type: "square",
    },
  }).chain(new Volume(-25), Destination)).current;`

4th way - values initialized once in useEffect having useState
 `const [mediumVolume, setMediumVolume] = useState<Volume | null>(null);
  const [sampler, setSampler] = useState<Sampler | null>(null);
  const [polySynthSquare, setPolySynthSquare] = useState<PolySynth | null>();

  useEffect(() => {
    console.log('useeffect');
    setMediumVolume(new Volume(-15));
    setPolySynthSquare(new PolySynth(Synth, {
      oscillator: {
        type: "square",
      },
    }).chain(new Volume(-25), Destination));
  }, []);

  useEffect(() => {
    console.log('useeffect-----2');
    if (mediumVolume) {
      setSampler(new Sampler({
        E4: DogSample,
      }).chain(mediumVolume, Destination));
    }
  }, [mediumVolume]);`
  
  5th way - values initialized once in useEffect having useRef
  `const mediumVolume = useRef<Volume | null>(null);
  const sampler = useRef<Sampler | null>(null);
  const polySynthSquare = useRef<PolySynth | null>();

  useEffect(() => {
    console.log('useeffect');
    mediumVolume.current = new Volume(-15);

    sampler.current = new Sampler({
      E4: DogSample,
    }).chain(mediumVolume.current, Destination);

    polySynthSquare.current = new PolySynth(Synth, {
      oscillator: {
        type: "square",
      },
    }).chain(new Volume(-25), Destination);
  }, []);`

Reply:
I think you've nailed it with #6 for your use case, it's by far the simplest and perfect for your app.

I think the only reason you'd want to look into any of the approaches would be if your app needed to be able to initialize/re-initialize the sound libraries based on the components' render lifecycle.

IF what you're looking for is a place to centralize your sampler/volume logic, then maybe what you're looking for is a central place to maintain that state, and a way to share/manipulate that state? If that's the case, you may be [in React.Context territory](https://github.com/laurasilvanavi/test-react-sounds-ts/pull/2#discussion_r821136779)

I'd also wonder if there's a different way to maintain state/share instances using tone.js, but unfortunately I'm a bit unfamiliar with that lib as well 😓